other/get.webgl.org/webgl2/index.html

```diff
@@ -182,18 +182,44 @@
     addClass($$("no-javascript"), "webgl-hidden");
     b = BrowserDetect;
     b.init();
+    var badImpl = false;
     var canvas = document.getElementById("webgl-logo");
     try { gl = canvas.getContext("webgl2"); }
     catch (x) { gl = null; }
 
+    if (gl) {
+        // check if it really supports WebGL2. Issues, Some browers claim to support WebGL2
+        // but in reality pass less than 20% of the conformance tests. Add a few simple
+        // tests to fail so as not to mislead users.
+        var params = [
+            { pname: 'MAX_3D_TEXTURE_SIZE', min: 256, },
+            { pname: 'MAX_DRAW_BUFFERS', min:4, },
+            { pname: 'MAX_COLOR_ATTACHMENTS', min:4, },
+            { pname: 'MAX_VERTEX_UNIFORM_BLOCKS', min:12, },
+            { pname: 'MAX_VERTEX_TEXTURE_IMAGE_UNITS', min:16, },
+            { pname: 'MAX_FRAGMENT_INPUT_COMPONENTS', min:60, },
+            { pname: 'MAX_UNIFORM_BUFFER_BINDINGS', min:24, },
+            { pname: 'MAX_COMBINED_UNIFORM_BLOCKS', min:24, },
+        ];
+        for (var i = 0; i < params.length; ++i) {
+            var param = params[i];
```

Comment:
Generally I'd like `for (const param of params)` these days, if you're not using `i`.

Reply:
I assumed since the rest of the code is using `var` that you wanted old compatible code. for of is pretty new and this site is supposed to run on old browsers to tell you to upgrade your browser.

Reply:
Fair! If this were only executed after checking for webgl2 support, we could rely on ES6.